src/Farmer/Common.fs

```diff
@@ -598,9 +598,20 @@ module Sql =
             | PremiumPool c ->
                 c
 
+module ManagedIdentity =
+    /// A user assigned managed identity that can be associated with a resource.
+    type UserAssignedIdentity =
+        | UserAssignedIdentity of CoreTypes.ResourceId
+        member this.ResourceId = match this with UserAssignedIdentity rId -> rId
+    type ResourceIdentity =
+        | SystemAssigned
+        | UserAssigned of UserAssignedIdentity list
```

Comment:
I was thinking of putting the `PrincipalId` member at this level but that doesn't work since User Assigned could be a list. Hmmm.